samples/matter/lock/src/combine_ota_images.py

```diff
@@ -0,0 +1,158 @@
+class ParseHeaderArgs:
+    def __init__(self, imageFile,):
+        self.image_file = imageFile
+
+class RemoveHeaderArgs:
+    def __init__(self, imageFile, outputFile):
+        self.image_file = imageFile
+        self.output_file = outputFile
+
+tlvHeaderKeysMap = {
+    ota.HeaderTag.VENDOR_ID : 'vendor_id',
+    ota.HeaderTag.PRODUCT_ID : 'product_id',
+    ota.HeaderTag.VERSION : 'version',
+    ota.HeaderTag.VERSION_STRING : 'version_str',
+    ota.HeaderTag.MIN_VERSION : 'min_version',
+    ota.HeaderTag.MAX_VERSION : 'max_version',
+    ota.HeaderTag.RELEASE_NOTES_URL : 'release_notes',
+    ota.HeaderTag.DIGEST_TYPE : 'digest_algorithm'
+}
+
+class GenerateOtaImageArgs:
+    def __init__(self, tlv: dict, input_file: str, output_file: str):
+        for key in tlvHeaderKeysMap:
+            valueToSet = tlv[key] if key in tlv else None
+            if tlvHeaderKeysMap[key] == 'digest_algorithm':
+                valueToSet = next(key for key, value in ota.DIGEST_ALGORITHM_ID.items() if value == valueToSet) # inverse lookup
+            self.__dict__[tlvHeaderKeysMap[key]] = valueToSet
+        self.input_files = [input_file]
+        self.output_file = output_file
+
+
+# Helper functions
+def remove_ota_header(imageFilePath: str) -> bytes:
+    """
+    Return binary representation of OTA DFU image file with Matter specific OTA header removed
+    """
+    with tempfile.TemporaryDirectory() as tempDir:
+        outputFile = os.path.join(tempDir, 'outputFile')
+        args = RemoveHeaderArgs(imageFilePath, outputFile)
+        ota.remove_header(args)
+        with open(args.output_file, 'rb') as extractedImageFile:
+            return extractedImageFile.read()
+
+def parse_multi_image_header(image: bytes) -> object:
+    """
+    Return JSON representation of NRF multi image DFU header extracted from input bytes object
+    """
+    with tempfile.TemporaryDirectory() as tempDir:
+        outputFile = os.path.join(tempDir, 'outputFile')
+        with open(outputFile, 'wb') as tempImgFile:
+            tempImgFile.write(image)
+        with open(outputFile, 'rb') as tempImgFile:
+            return multi.parse_header(tempImgFile)
+
+def extract_core_images(fullImage: bytes) -> list:
+    """
+    Extract app core and net core raw images from given NRF multi image binary representation
+    """
+    cborHeader = parse_multi_image_header(fullImage)
+    APP_CORE_ID = 0
+    offset = cborHeader["img"][APP_CORE_ID]['size']
+    bytesHeader = cbor2.dumps(cborHeader)
+    imageStart = len(bytesHeader) + len(struct.pack('<H', len(bytesHeader))) # the header is preceded by its size (encoded on 2 bytes)
+    headerLessImage = fullImage[imageStart:]
+    return headerLessImage[:offset], headerLessImage[offset:]
+
+def tlv_headers_equal(first: dict, second: dict) -> bool:
+    """
+    Verify if the relevant TLV header fields are the same across two provided JSONs
+    """
+    keys_to_compare = [ota.HeaderTag.VENDOR_ID, ota.HeaderTag.PRODUCT_ID,
```

Comment:
I think we should compare min/max version and release_notes_url as well. Probably better to define `exclude_list` that would only contain `digest` and `payload_size`.

Reply:
Ok, will fix that.